### PR TITLE
IDENTITY-3561 Self Sign Up and Account Confirmation - Generated Confirmation Keys Not Get Deleted

### PR DESCRIPTION
**Pull request by cdwijayarathna**

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 1]
```diff
@@ -34,6 +34,7 @@
         <RememberMeTimeout>20160</RememberMeTimeout>
         <PersistanceCleanUpTimeout>20160</PersistanceCleanUpTimeout>
         <PersistanceCleanUpPeriod>1140</PersistanceCleanUpPeriod>
+        <RegistryCleanUpPeriod>1440</RegistryCleanUpPeriod>
```
Comment (johannnallathamby) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Shall we move this to identity-mgt.properties ? The name here is too generic, it doesn't indicate anything about identity-mgt. May be in next release, once we move all the configurations to UI, the remaining configurations in identity-mgt.properties we can put it to identity.xml and remove that file.

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                    } catch (RegistryException e) {
+                        log.error("Error while deleting the expired confirmation code.", e);
+                    }
+                    PrivilegedCarbonContext.endTenantFlow();
+                }
+            } catch (UserStoreException e) {
+                log.error("Error while getting the tenant manager.", e);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Stop running the Identity-Management registry Data cleanup task.");
```
Comment (johannnallathamby) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java]: "Stop" should be changed as "Finished"

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                                    String[] secondaryStoreResourcePaths = secondaryStoreCollection.getChildren();
+                                    for (int k = 0; k < secondaryStoreResourcePaths.length; k++) {
+
+                                        checkAndDeleteRegistryResource(registry, secondaryStoreResourcePaths[k]);
+                                    }
+                                } else {
+                                    checkAndDeleteRegistryResource(registry, identityResourcesPaths[j]);
+                                }
+                            } catch (RegistryException e) {
+                                log.error("Error while retrieving resource at " + identityResourcesPaths[j]);
```
Comment (johannnallathamby) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java]: add stack trace ", e"

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                            }
+                        }
+                    } catch (ResourceNotFoundException e) {
+                        log.error("No resource found for tenant " + tenant.getDomain(), e);
+                    } catch (RegistryException e) {
+                        log.error("Error while deleting the expired confirmation code.", e);
+                    }
+                    PrivilegedCarbonContext.endTenantFlow();
```
Comment (johannnallathamby) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java]: This must go to finally block

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                        identityDataResource = (Collection) registry.get(CONFIRMATION_REGISTRY_RESOURCE_PATH);
+                        String[] identityResourcesPaths = identityDataResource.getChildren();
+                        for (int j = 0; j < identityResourcesPaths.length; j++) {
+                            try {
+                                Resource currentResource = registry.get(identityResourcesPaths[j]);
+                                if (currentResource instanceof Collection) {
+                                    Collection secondaryStoreCollection = (Collection) currentResource;
+                                    String[] secondaryStoreResourcePaths = secondaryStoreCollection.getChildren();
+                                    for (int k = 0; k < secondaryStoreResourcePaths.length; k++) {
+
```
Comment (pulasthi7) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java]: unnecessary new line

Comment (cdwijayarathna) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryCleanUpService.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1675

**Pull request merged**